hashes/src/internal_macros.rs

```diff
@@ -199,18 +199,30 @@ macro_rules! hash_type {
     ($bits:expr, $reverse:expr, $doc:literal, $schemars:literal) => {
         #[doc = $doc]
         #[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
-        #[cfg_attr(feature = "schemars", derive(crate::schemars::JsonSchema))]
         #[repr(transparent)]
-        pub struct Hash(
-            #[cfg_attr(feature = "schemars", schemars(schema_with = $schemars))] [u8; $bits / 8],
-        );
+        pub struct Hash([u8; $bits / 8]);
 
         impl Hash {
             fn internal_new(arr: [u8; $bits / 8]) -> Self { Hash(arr) }
 
             fn internal_engine() -> HashEngine { Default::default() }
         }
 
+        #[cfg(feature = "schemars")]
+        impl schemars::JsonSchema for Hash {
+            fn schema_name() -> String { "Hash".to_owned() }
+
+            fn json_schema(gen: &mut schemars::gen::SchemaGenerator) -> schemars::schema::Schema {
+                match $bits / 8 {
+                    8 => crate::util::json_hex_string::len_8(gen),
+                    20 => crate::util::json_hex_string::len_20(gen),
+                    32 => crate::util::json_hex_string::len_32(gen),
+                    64 => crate::util::json_hex_string::len_64(gen),
+                    _ => unreachable!("we only ever call this macro with theses sizes"),
```

Comment:
If we're not using the derive anyway there's no point of calling `len_*` functions which have the parameters hardcoded. Just inline them here and use bits/8 instead of `$len` directly. Also we can remove the `$schemars` parameter from macro.

Reply:
Thanks man, will re-spin.